app/src/main/java/org/oppia/app/walkthrough/WalkthroughActivityPresenter.kt

```diff
@@ -1,25 +1,69 @@
 package org.oppia.app.walkthrough
 
+import android.widget.ImageView
+import android.widget.ProgressBar
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.Observer
 import org.oppia.app.R
 import org.oppia.app.activity.ActivityScope
+import org.oppia.app.walkthrough.end.WalkthroughFinalFragment
+import org.oppia.app.walkthrough.topiclist.WalkthroughTopicListFragment
+import org.oppia.app.walkthrough.welcome.WalkthroughWelcomeFragment
 import javax.inject.Inject
 
 /** The presenter for [WalkthroughActivity]. */
 @ActivityScope
 class WalkthroughActivityPresenter @Inject constructor(private val activity: AppCompatActivity) {
+  private var currentProgress = MutableLiveData(0)
```

Comment:
Instead of this, introduce a ViewModel and use that to control the progress using an `ObservaleField<Int>`

Reply:
done